processor/src/main/java/org/mapstruct/ap/internal/model/MappingMethod.java

```diff
@@ -238,9 +238,6 @@ public boolean equals(Object obj) {
             return false;
         }
         final MappingMethod other = (MappingMethod) obj;
-        if ( (this.name == null) ? (other.name != null) : !this.name.equals( other.name ) ) {
```

Comment:
Comment in the code?

The problem is that if take name into consideration and we always generate safe names then 2 forge methods will always be different. If I didn't remove it then the test for the `DomainDtoWithNvmsDefaultMapper` was failing because we were not reusing the forged (not via automapping, but for `Iterables`) `protected Set<Long> stringListToLongSet(List<String> list)`.

This is connected with adding the `super.equals( obj )` in the `NormalTypeMappingMethod`. What I can do is to add the `parameters` and `returnType` to the `equals` of the `NormalTypeMappingMethod`